### PR TITLE
fix(atomic): center refine modal contents

### PR DESCRIPTION
**Pull request by samisayegh**
I started with [this simpler solution](https://github.com/coveo/ui-kit/pull/1162/commits/4d7bbf260e7f08217ac76d0540b8846096a769f0), but it does not work outside of Chrome, which is the only browser that supports `overflow-y: overlay`.

This approach uses a negative margin to shift the contents and adjust for the scroll bar.

**Screen**

refine-modal_centered_chrome

https://coveord.atlassian.net/browse/KIT-957

### CODE REVIEW

File: packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss [hunk 1]
```diff
@@ -11,3 +11,11 @@ select:hover + div,
 select:focus + div {
   @apply text-primary-light;
 }
+
+.centered {
+  @apply w-full max-w-lg mx-auto;
```
Comment (samisayegh) [inline packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss:16]: `max-w-lg` corresponds to `512px`.

File: packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss [hunk 2]
```diff
@@ -11,3 +11,11 @@ select:hover + div,
...
+}
+
+.adjust-for-scroll-bar {
+  margin-right: calc(-1 * (100vw - 3rem - 100%));
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss:20]: Why is this needed, exactly? When I disable it locally on Chrome and Firefox, the scrollbar isn't affected.

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss (hunk 2 above)
Comment (olamothe) [reply packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss:20]: I think this would benefit from a comment/explanation about what it does and why it is needed.

It's a pretty arcane bit of CSS :D

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss (hunk 2 above)
Comment (samisayegh) [reply packages/atomic/src/components/atomic-refine-modal/atomic-refine-modal.pcss:20]: I created [a video](https://share.vidyard.com/watch/QFRrnPkuRd6UToirD91zDC?) where I toggle this line on and off. When off, the middle panel is offset from the header and footer by the width of the scroll bar.
